src/m365/aad/commands/administrativeunit/administrativeunit-add.spec.ts

```diff
@@ -0,0 +1,192 @@
+import assert from 'assert';
+import sinon from 'sinon';
+import auth from '../../../../Auth.js';
+import { Cli } from '../../../../cli/Cli.js';
+import { CommandInfo } from '../../../../cli/CommandInfo.js';
+import commands from '../../commands.js';
+import command from './administrativeunit-add.js';
+import { telemetry } from '../../../../telemetry.js';
+import { pid } from '../../../../utils/pid.js';
+import { session } from '../../../../utils/session.js';
+import { sinonUtil } from '../../../../utils/sinonUtil.js';
+import request from '../../../../request.js';
+import { Logger } from '../../../../cli/Logger.js';
+import { CommandError } from '../../../../Command.js';
+
+describe(commands.ADMINISTRATIVEUNIT_ADD, () => {
+  const administrativeUnitReponse: any = {
+    id: 'fc33aa61-cf0e-46b6-9506-f633347202ab',
+    displayName: 'European Division',
+    description: null,
+    visibility: null
+  };
+
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+    (command as any).pollingInterval = 0;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.post
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.ADMINISTRATIVEUNIT_ADD);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('creates an administrative unit with a specific display name', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === 'https://graph.microsoft.com/v1.0/directory/administrativeUnits') {
+        return administrativeUnitReponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { displayName: 'European Division' } });
+    assert.deepStrictEqual(postStub.lastCall.args[0].data, {
+      displayName: 'European Division',
+      description: undefined,
+      visibility: null
+    });
+    assert(loggerLogSpy.calledOnceWith(administrativeUnitReponse));
```

Comment:
Let's use `calledOnceWithExactly`

Reply:
Used `calledOnceWithExactly` instead of `calledOnceWith`